internal/pubsub/pubsub_test.go

```diff
@@ -63,11 +63,11 @@ func NewDriverSub() *driverSub {
 	return ds
 }
 
-func (s *driverSub) ReceiveBatch(ctx context.Context) ([]*driver.Message, error) {
+func (s *driverSub) ReceiveBatch(ctx context.Context, maxMessages int) ([]*driver.Message, error) {
 	for {
 		select {
 		case <-s.sem:
-			ms := s.grabQueue()
+			ms := s.grabQueue(maxMessages)
```

Comment:
This loop is going to burn CPU if len(s.q) == 0. Maybe a small sleep?

Reply:
(Unrelated to this PR; you can fix in a different one.)

Reply:
Ok